pkg/server/server.go

```diff
@@ -139,15 +144,10 @@ func (k *KiamServer) GetPodRole(ctx context.Context, req *pb.GetPodRoleRequest)
 	}
 
 	role := k8s.PodRole(pod)
-	ref, err := reference.GetReference(scheme.Scheme, pod)
-	if err != nil {
-		logger.Errorf("error getting reference for pod %q: %s", pod.Name, err)
-		return nil, err
-	}
 
 	logger.WithField("pod.iam.role", role).Infof("found role")
-	k.eventRecorder.Event(ref, v1.EventTypeNormal, "KiamRoleFound",
-		fmt.Sprintf("Role: %q found for pod: %q", role, pod.Name))
+	k.recordEvent(pod, v1.EventTypeNormal, "KiamRoleFound",
```

Comment:
As above, I'd want to know we could do this without risk of DDoS'ing the API server and etcd- I think these endpoints will get called a lot.